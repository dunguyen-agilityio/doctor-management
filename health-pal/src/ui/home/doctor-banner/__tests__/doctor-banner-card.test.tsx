import { render, screen } from '@utils-test'

import DoctorBannerCard from '../doctor-banner-card'

describe('DoctorBannerCard', () => {
  const defaultProps = {
    title: 'Top Doctors',
    description: 'Find the best doctors near you',
  }

  it('renders default image and text content', () => {
    render(<DoctorBannerCard {...defaultProps} />)

    expect(screen.getByText('Top Doctors')).toBeTruthy()
    expect(screen.getByText('Find the best doctors near you')).toBeTruthy()
    expect(screen.getByTestId('banner-image')).toBeTruthy()
  })

  it('uses custom image when provided', () => {
    const customImage = 'https://example.com/banner.png'

    render(<DoctorBannerCard {...defaultProps} image={customImage} />)

    const image = screen.getByTestId('banner-image')
    expect(image).toHaveProp(
      'source',
      expect.arrayContaining([{ uri: 'https://example.com/banner.png' }]),
    )
  })

  it('matches snapshot', () => {
    const { toJSON } = render(<DoctorBannerCard {...defaultProps} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
