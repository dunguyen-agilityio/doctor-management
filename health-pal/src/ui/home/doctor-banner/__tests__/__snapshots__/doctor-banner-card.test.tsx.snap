// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorBannerCard matches snapshot 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "flexDirection": "column",
        "gap": 8,
        "overflow": "hidden",
        "paddingLeft": 12,
        "paddingTop": 32,
        "position": "relative",
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      inset={0}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={[]}
      position="absolute"
      source={
        [
          {
            "testUri": "../../../assets/images/banner01.webp",
          },
        ]
      }
      style={
        {
          "inset": 0,
          "position": "absolute",
        }
      }
      testID="banner-image"
      transition={null}
    />
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Inter_700Bold",
          "fontSize": 16,
          "lineHeight": 24,
        }
      }
      suppressHighlighting={true}
    >
      Top Doctors
    </Text>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Inter_400Regular",
          "fontSize": 12,
          "lineHeight": 18,
        }
      }
      suppressHighlighting={true}
    >
      Find the best doctors near you
    </Text>
  </View>,
]
`;
